### ISSUE TITLE
building external connector

### ISSUE THREAD

**Issue by sehz** [MaterializeInc/materialize#6455]
Hi, [Fluvio Project](https://github.com/infinyon/fluvio) is interested in building a connector for Materialize as some of our users are interested in using it to query data stored in our platform. 

However, it seems there is no way to build a third-party connector without modifying Materialize's codebase.   Is there a documented way or API that lets us create a connector that can be integrated into the installed Materialized binary?

**Comment by umanwizard** [comment]
No, we don’t have any such feature in Materialize now, but we are definitely interested in scoping one out in the future.

Can you say a bit more about what Fluvio is, how data is stored there, and how you envision streaming it to Materialize?

**Comment by sehz** [comment]
We are distributed stream platform.  You can read more about it [here](https://www.fluvio.io/docs/).  We have an append-only immutable data store, which can be replicated.   I would imagine it would be similar to your `Kafka` or `Kinesis` data source/sink.  They can be mapped to our [produce/consumer](https://docs.rs/fluvio/0.6.1/fluvio/) API.

**Comment by umanwizard** [comment]
Does Fluvio have the notion of consistency or transaction boundaries across different streams that Materialize would need to respect?

**Comment by sehz** [comment]
We have a durable consistency guarantee (how many replicas must be synced) that the client configure.  We don't have transaction boundaries.
